sql/moz-fx-data-shared-prod/bigeye_derived/virtual_table_service_v1/query.py

```diff
@@ -0,0 +1,141 @@
+    }
+
+    response = requests.get(API_URL + str(workspace_id), headers=headers)
+    response.raise_for_status()
+    return response.json()
+
+
+def process_response(response_data: Dict[str, Any]) -> pd.DataFrame:
+    """Process API response into a DataFrame."""
+    df = pd.json_normalize(
+        response_data,
+        record_path=["table", "columns"],
+        meta=[
+            ["table", "id"],
+            ["table", "schemaName"],
+            ["table", "name"],
+            ["table", "warehouseId"],
+            ["table", "latestMetricTimeValue"],
+            ["table", "schemaId"],
+            ["table", "popularityScore"],
+            ["table", "warehouseName"],
+            ["table", "warehouseVendor"],
+            ["table", "dataNodeId"],
+            ["table", "numQueriesLatestPeriod"],
+            ["table", "tableType"],
+            ["table", "isFavorite"],
+            ["table", "requiresPartitionFilter"],
+            ["table", "usesAgent"],
+            ["sqlQuery"],
+            "tags",
+            ["table", "schema", "id"],
+            ["table", "schema", "name"],
+            ["table", "schema", "warehouseId"],
+            ["table", "schema", "isVirtual"],
+            ["table", "schema", "popularityScore"],
+            ["table", "schema", "numQueriesLatestPeriod"],
+            ["table", "schema", "dataNodeId"],
+            ["table", "schema", "warehouseName"],
+            ["table", "schema", "databaseType"],
+            ["table", "schema", "indexByTable"],
+            ["table", "schemaChangeInfo", "mostRecentSchemaScanAt"],
+            ["table", "schemaChangeInfo", "initialSchemaScanAt"],
+            ["table", "schema", "schemaChangeInfo", "lastSchemaChangeAt"],
+            ["table", "schema", "schemaChangeInfo", "initialSchemaScanAt"],
+        ],
+        meta_prefix="table_",
+        sep="_",
+        errors="ignore",
+    )
+
+    df["id"] = df["id"].astype("Int64")
+    df["schemaChangeInfo_lastSchemaChangeAt"] = df[
+        "schemaChangeInfo_lastSchemaChangeAt"
+    ].astype("Int64")
+    df["schemaChangeInfo_mostRecentSchemaScanAt"] = df[
+        "schemaChangeInfo_mostRecentSchemaScanAt"
+    ].astype("Int64")
+    df["schemaChangeInfo_initialSchemaScanAt"] = df[
+        "schemaChangeInfo_initialSchemaScanAt"
+    ].astype("Int64")
+
+    df["refreshed_at"] = pd.Timestamp.now(tz="UTC")
+
+    return df
+
+
+def get_bigeye_data() -> pd.DataFrame:
+    """Fetch data from Bigeye API for all workspaces and combine into a single DataFrame."""
+    all_data = []
+
+    if not isinstance(WORKSPACE_IDS, list):
+        raise Exception("WORKSPACE_IDS is not a list")
+
+    for workspace_id in WORKSPACE_IDS:
+        try:
+            response_data = make_api_request(workspace_id)
+            df = process_response(response_data)
+            all_data.append(df)
+            logging.info(f"Successfully processed data for workspace {workspace_id}")
+        except Exception as e:
```

Comment:
Maybe as a future improvement we should consider using a more specific exception here so that we don't just catch everything here making it more difficult to understand what the actual underlying issues is.